docs/configuring_and_running_tests.md

```diff
@@ -500,6 +506,40 @@ If you have multiple test programs, such as both a client and a server, or even
 different sets of arguments, you should name the relevant config YAML and known-failing files so
 it is clear to which invocation they apply.
 
+## Upgrading
+
+When a new version of the conformance suite is released, ideally, you could simply update
+the version number you are using and everything just works. We aim for
+backwards-compatibility between releases to maximize the chances of this ideal outcome. But
+there are a number of things that can happen in a release that make the process a little
+more laborious:
+
+* As a matter of hygiene/maintenance, we may rename and re-organize test suites and test
+  cases. This means that any test case patterns that are part of your configuration (like
+  known-failing files) may need to be updated. We don't expect this to happen often, but
+  when it does, we will include information in the release notes to aid in updating your
+  configuration.
+* The new version may contain new/updated test cases that require some changes in the
+  behavior/logic of your implementations under test. This might be for testing new
+  functionality that requires new fields in the conformance protocol messages. Without
+  changes in your client or server under test, the new test cases will likely fail.
+* The new version may contain new/updated test cases that reveal previously undetected
+  conformance failures.
+
+To minimize disruption when upgrading, we recommend a process that looks like so:
+1. Update to the new release of the conformance suite.
+2. Update test case patterns (like in known-failing configurations) if necessary to match
+   any changes to test case names and organization.
+3. Update/add known-failing configurations for any new failures resulting from new/updated
```

Comment:
This is one case where automatically appending tests to a `known-failing` file would be super helpful.